tuist/tuist
# Ensure precompiled frameworks of target applications aren't included in UI test targets

Resolves: https://github.com/tuist/tuist/issues/887

### Short description 📝

- Test targets declare their host applications as a dependency
- This lead to a few graph queries transitively including depdencies of the application in the test targets

### Solution 📦

- When performing dependency searches, we can skip dependency tree of nodes that can embed products (e.g. Applications)

### Implementation 👩‍💻👨‍💻

- [x] Update Graph `embeddableDependencies` to skip application targets when traversing test dependencies 
- [x] Add acceptance test

### Test Plan 🛠:
- Run `tuist generate` within `fixtures/ios_app_with_transitive_framework`
- Open the generated workspace
- Verify that `AppUITests` has no precompiled frameworks included
